pgjdbc/src/main/java/org/postgresql/core/v3/QueryExecutorImpl.java

```diff
@@ -351,11 +354,13 @@ private boolean sendAutomaticSavepoint(Query query, int flags) throws IOExceptio
         // If CompositeQuery is observed, just assume it might fail and set the savepoint
         || !(query instanceof SimpleQuery)
         || ((SimpleQuery) query).getFields() != null)) {
-      sendOneQuery(autoSaveQuery, SimpleQuery.NO_PARAMETERS, 1, 0,
-          updateQueryMode(QUERY_NO_RESULTS | QUERY_NO_METADATA)
-              // PostgreSQL does not support bind, exec, simple, sync message flow,
-              // so we force autosavepoint to use simple if the main query is using simple
-              | (flags & QueryExecutor.QUERY_EXECUTE_AS_SIMPLE));
+      EnumSet<QueryFlag> flags2 = EnumSet.copyOf(flags);
+      flags2.add(QueryFlag.NO_RESULTS);
+      flags2.add(QueryFlag.NO_METADATA);
+      // PostgreSQL does not support bind, exec, simple, sync message flow,
+      // so we force autosavepoint to use simple if the main query is using simple
+      flags2.add(QueryFlag.EXECUTE_AS_SIMPLE);
```

Comment:
_if the main query is using simple_ part is missing, could be expressed like this:
```
EnumSet<QueryFlag> flags2 = EnumSet.copyOf(flags);
flags2.retainAll(EnumSet.of(EXECUTE_AS_SIMPLE))
flags2.add(NO_RESULTS)
flags2.add(NO_METADATA)
```